### PR TITLE
setup guide

### PR DESCRIPTION
**Pull request by melrchen**

### CODE REVIEW

File: setup-guide.mdx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+collaborate efficiently and effectively.
+
+In this tutorial, we will guide you step by step through customizing your Chalk solution to help you achieve all of
+your data goals.
+
+---
+
+## Creating your Chalk Project
+
+1. If you don't already have a Github repository in which you will store your Chalk code, create one. Then, pick a
```
Comment (hugokitano) [inline setup-guide.mdx:20]:
```suggestion
1. If you don't already have a Github repository where you will store your Chalk code, create one. Then, pick a
```

File: setup-guide.mdx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+Once you have deployed your code, then you can query your features directly from the command line using the
+[`chalk query`](https://docs.chalk.ai/cli/query) command or from within a notebook using the
+[ChalkClient](https://docs.chalk.ai/api-docs#ChalkClient) to run `ChalkClient.query()` or `ChalkClient.offline_query()`.
+
+This is the primary workflow for iterating on features and resolvers! Write, deploy, and query to verify whether the
+feature values that you receive are the expected values. Once you have finalized your feature set and resolver
+definitions, then the final step is frequently orchestration.
+
+## Orchestration
```
Comment (emarx) [inline setup-guide.mdx:158]: above the double headers, the convention has been to insert a line break

```suggestion
---

## Orchestration
```

File: setup-guide.mdx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+├── chalk.yaml
+├── README.md
+└── requirements.txt
+```
+
+You can read more in our docs about the [different kinds of features](https://docs.chalk.ai/docs/features) and the
+[different kinds of resolvers](https://docs.chalk.ai/docs/resolver-overview) that you can write. If you would like
+guidance on how to structure your featuresets and resolvers, please reach out in your support channel!
+
+## Deploy and Query
```
Comment (emarx) [inline setup-guide.mdx:143]:
```suggestion
---

## Deploy and Query
```

File: setup-guide.mdx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+ ```
+
+Say we added a PostgreSQL data source, then our `datasources.py` might look something like this:
+```python
+pg = PostgreSQLSource(name='PG')
+```
+
+For more details on setting up data sources, [see here.](https://docs.chalk.ai/docs/integrations)
+
+##  Define Feature Sets and Resolvers
```
Comment (emarx) [inline setup-guide.mdx:113]:
```suggestion
---

##  Define Feature Sets and Resolvers
```

File: setup-guide.mdx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+The second file, `README.md`, contains some basic commands you can use with the Chalk CLI.
+The third file, `requirements.txt` should look something like this:
+   ```
+    requests
+    chalkpy[runtime]
+    ```
+
+   This contains your project dependencies.
+   You can add a file `.chalkignore` which will act just like a `.gitignore` file.
```
Comment (hugokitano) [inline setup-guide.mdx:70]: mention that it excludes from chalk apply?

File: setup-guide.mdx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+```bash Terminal
+$ pip3 install -r requirements.txt
+```
+
+Now, you should have a virtual environment with all project dependencies installed and a basic project structure upon
+which we will build in the next step.
+
+## Configure Datasources
+
+Having set up a basic project directory, we want to configure the data sources from which we will load data to
```
Comment (hugokitano) [inline setup-guide.mdx:81]:
```suggestion
Having set up a basic project directory, next we'll want to configure the data sources from which we will load data to
```

File: setup-guide.mdx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+compute our features. We can do so in the Chalk dashboard.
+
+1. In the same directory from before, login or signup with Chalk directly from the command line. The
+[`chalk login`](https://docs.chalk.ai/cli/login) command will open your browser and create an
+API token for your local development, as well as redirect you to your dashboard. If you are not
+redirected you can also find the dashboard at [https://chalk.ai/projects](https://chalk.ai/projects).
+You will automatically see all of the environments in which the email you used to login has been
```
Comment (hugokitano) [inline setup-guide.mdx:88]: you can also do `chalk dashboard`

File: setup-guide.mdx [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+provisioned as a user.
+ 2. Within the dashboard, navigate to `Data Sources` in the side bar and add all data sources that you will
+be working with here. After you have saved a data source, you can also use the `Test Data Source` button
```
Comment (hugokitano) [inline setup-guide.mdx:91]:
```suggestion
be working with here. After you have saved a data source, you can use the `Test Data Source` button
```

File: setup-guide.mdx [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+ ├── README.md
+ └── requirements.txt
+ ```
+
+Say we added a PostgreSQL data source, then our `datasources.py` might look something like this:
+```python
+pg = PostgreSQLSource(name='PG')
+```
+
+For more details on setting up data sources, [see here.](https://docs.chalk.ai/docs/integrations)
```
Comment (hugokitano) [inline setup-guide.mdx:111]:
```suggestion
For more details on setting up data sources, [see here](https://docs.chalk.ai/docs/integrations).
```

File: setup-guide.mdx [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+
+##  Define Feature Sets and Resolvers
+
+Having set up our data sources, we now want to define the registry for this feature store by writing our
```
Comment (hugokitano) [inline setup-guide.mdx:117]: maybe not repeat the same intro as the previous step? also, i dont think the concept of registry is user facing aka they wont know what this means

File: setup-guide.mdx [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+feature sets and resolvers. Each *feature set* is a Python class of features, and a *resolver* tells Chalk
```
Comment (hugokitano) [inline setup-guide.mdx:118]:
```suggestion
feature sets and resolvers. Each *feature set* is a Python class of features, and each *resolver* tells Chalk
```

File: setup-guide.mdx [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+│  └── ...
+├── .chalkignore
+├── chalk.yaml
+├── README.md
+└── requirements.txt
+```
+
+You can read more in our docs about the [different kinds of features](https://docs.chalk.ai/docs/features) and the
+[different kinds of resolvers](https://docs.chalk.ai/docs/resolver-overview) that you can write. If you would like
+guidance on how to structure your featuresets and resolvers, please reach out in your support channel!
```
Comment (hugokitano) [inline setup-guide.mdx:143]:
```suggestion
guidance on how to structure your feature sets and resolvers, please reach out in your support channel!
```

File: setup-guide.mdx [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+
+## Deploy and Query
+
+Having defined features and resolvers, the next step is to deploy your pipelines. You can deploy to production by
```
Comment (hugokitano) [inline setup-guide.mdx:149]: same as above, easy to gloss over sentences if they have the same structure as something you've already read

File: setup-guide.mdx [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+using the [`chalk apply`](https://docs.chalk.ai/cli/apply) command. During development, we recommend that you use
+`chalk apply --branch {BRANCH_NAME}` to deploy to the [branch server](https://docs.chalk.ai/docs/branches), which allows
+multiple people to work concurrently in one environment, and also enables more performant deploys.
+
+Once you have deployed your code, then you can query your features directly from the command line using the
+[`chalk query`](https://docs.chalk.ai/cli/query) command or from within a notebook using the
+[ChalkClient](https://docs.chalk.ai/api-docs#ChalkClient) to run `ChalkClient.query()` or `ChalkClient.offline_query()`.
```
Comment (hugokitano) [inline setup-guide.mdx:156]: maybe mention our other clients (go, java etc)?

File: setup-guide.mdx [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+
+This is the primary workflow for iterating on features and resolvers! Write, deploy, and query to verify whether the
+feature values that you receive are the expected values. Once you have finalized your feature set and resolver
```
Comment (hugokitano) [inline setup-guide.mdx:159]:
```suggestion
feature values that you receive are the values you expect. Once you have finalized your feature set and resolver
```

**Review by sjmignot — APPROVED** [review]

File: setup-guide.mdx [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+definitions, then the final step is frequently orchestration.
```
Comment (hugokitano) [inline setup-guide.mdx:160]:
```suggestion
definitions, the final step is frequently orchestration.
```

File: setup-guide.mdx [hunk 17]
```diff
@@ -0,0 +1,167 @@
...
+
+## Orchestration
+
+Once you have verified that your feature and resolver definitions are correct, then the next step is to determine how
```
Comment (hugokitano) [inline setup-guide.mdx:166]:
```suggestion
Once you have verified that your feature and resolver definitions are correct, the next step is to determine how
```

File: setup-guide.mdx [hunk 18]
```diff
@@ -0,0 +1,167 @@
...
+you want to use the corresponding feature values within your larger machine learning platform. Some users [trigger
+resolvers](https://docs.chalk.ai/docs/runs) and run queries from within other orchestrated pipelines, such as Airflow.
+Some users define [cron schedules](https://docs.chalk.ai/docs/resolver-cron)for their resolvers,
+and [set staleness values](https://docs.chalk.ai/docs/feature-caching) on different features to
+ensure that the data they query falls within their requirements for freshness. The data world is your oyster!
+
+But, as always, if you would like guidance on how to configure that oyster, please reach out in your support channel!
```
Comment (hugokitano) [inline setup-guide.mdx]: lol

**Review by hugokitano — APPROVED** [review]

File: setup-guide.mdx (hunk 7 above)
Comment (melrchen) [reply setup-guide.mdx:88]: TIL

File: setup-guide.mdx (hunk 13 above)
Comment (melrchen) [reply setup-guide.mdx:149]: facts

File: setup-guide.mdx (hunk 14 above)
Comment (melrchen) [reply setup-guide.mdx:156]: ok true

**Pull request merged**